app/src/main/java/org/asteroidos/sync/connectivity/IConnectivityService.java

```diff
@@ -0,0 +1,39 @@
+package org.asteroidos.sync.connectivity;
+
+import java.util.HashMap;
+import java.util.UUID;
+
+/**
+ * A connectivity service is a module that can exchange data with the watch. It has to implement {@link IService}
+ * and additional functions regarding connectivity from {@link IConnectivityService}.
+ */
+public interface IConnectivityService extends IService {
+
+    enum Direction{
+        RX,
+        TX
+    }
+
+    public HashMap<UUID, Direction> getCharacteristicUUIDs();
+
+    public UUID getServiceUUID();
+
+    /**
+     * <code>onReceive()</code> is called when data from the watch is received.
+     *
+     * <pre>
+     * &#064;Override
+     * <b>public void</b> onReceive(UUID uuid, byte[] data) {
+     *     //handle data
+     * }
+     * </pre>
+     *
+     * @param uuid {@link UUID} from {@link org.asteroidos.sync.utils.AsteroidUUIDS}
+     * @param data payload for the {@link IConnectivityService}
+     *
+     * @return <code>true</code> if the data was processed by the {@link IConnectivityService},
+     *     <code>false</code> if the data was intentionally not used (e.g. the {@link IConnectivityService} does not implement
+     *     a method to process the data)
+     */
+    public Boolean onReceive(UUID uuid, byte[] data);
```

Comment:
It is awkward that all services get called when something is received. And most services don't even do anything with it.

I've been thinking about this a bit and I think that there should be two hashmaps instead off gattChars.

For the RX case, services should have a way to register callback**s** for all UUIDs they are interested in, instead of just indicating that they have RX characteristics. A hashmap could keep track of the UUID -> callback correspondance. For services like the media control, it would keep each callback separate and avoid having to check UUIDs all the time.

For TX, I think there should be a distinct UUID -> BluetoothGattCharacteristic hashmap. Currently on the send path you you iterate over gattChars linearly, compare each UUID and discard the direction.